server/src/main/java/org/elasticsearch/action/admin/cluster/storedscripts/GetStoredScriptResponse.java

```diff
@@ -21,39 +21,98 @@
 
 import org.elasticsearch.Version;
 import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.common.ParseField;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.StatusToXContentObject;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.rest.RestStatus;
 import org.elasticsearch.script.StoredScriptSource;
 
 import java.io.IOException;
+import java.util.Objects;
 
-public class GetStoredScriptResponse extends ActionResponse implements ToXContentObject {
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
 
+public class GetStoredScriptResponse extends ActionResponse implements StatusToXContentObject {
+
+    public static final ParseField _ID_PARSE_FIELD = new ParseField("_id");
+    public static final ParseField FOUND_PARSE_FIELD = new ParseField("found");
+    public static final ParseField SCRIPT = new ParseField("script");
+
+    private static final ConstructingObjectParser<GetStoredScriptResponse, String> PARSER =
+        new ConstructingObjectParser<>("GetStoredScriptResponse",
+            true,
+            (a, c) -> {
+                String id = (String) a[0];
+                boolean found = (Boolean)a[1];
+                StoredScriptSource scriptSource = (StoredScriptSource)a[2];
+                return found ? new GetStoredScriptResponse(id, scriptSource) : new GetStoredScriptResponse(id);
+            });
+
+    static {
+        PARSER.declareField(constructorArg(), (p, c) -> p.text(),
+            _ID_PARSE_FIELD, ObjectParser.ValueType.STRING);
+        PARSER.declareField(constructorArg(), (p, c) -> p.booleanValue(),
+            FOUND_PARSE_FIELD, ObjectParser.ValueType.BOOLEAN);
+        PARSER.declareField(optionalConstructorArg(), (p, c) -> StoredScriptSource.fromXContent(p, true),
+            SCRIPT, ObjectParser.ValueType.OBJECT);
+    }
+
+    private String id;
     private StoredScriptSource source;
 
     GetStoredScriptResponse() {
     }
 
-    GetStoredScriptResponse(StoredScriptSource source) {
+    GetStoredScriptResponse(String id) {
```

Comment:
maybe we can do without this constructor and always use the other that accepts both id and source? otherwise make this one private at least?